render/canvas/src/lib.rs

```diff
@@ -103,12 +107,145 @@ impl CanvasFillStyle {
     }
 }
 
+/// Stores the actual bitmap data on the browser side in one of two ways.
+/// Each better suited for different scenarios and source data formats.
+/// ImageBitmap could unify these somewhat, but Safari doesn't support it.
+enum BitmapDataStorage {
+    /// Utilizes the JPEG decoder of the browser, and can be drawn onto a canvas directly.
+    /// Needs to be drawn onto a temporary canvas to retrieve the stored pixel data.
+    ImageElement(HtmlImageElement),
+    /// Much easier to create from raw RGB[A] data, through a temporary ImageData.
+    /// The pixel data can also be retrieved through a temporary ImageData.
+    CanvasElement(HtmlCanvasElement, CanvasRenderingContext2d),
+}
+
+impl BitmapDataStorage {
+    /// Puts the image data into a newly created <canvas>, and caches it.
+    fn from_image_data(data: ImageData) -> Self {
+        let window = web_sys::window().unwrap();
+        let document = window.document().unwrap();
+
+        let canvas: HtmlCanvasElement = document
+            .create_element("canvas")
+            .unwrap()
+            .dyn_into()
+            .unwrap();
+
+        canvas
```

Comment:
the element isn't attached to the DOM, right? Does setting `display` do anything here?

Reply:
Hmm, good catch, it most likely doesn't do a thing.
TBH I mostly just copied what's in `WebCanvasRenderBackend::push_render_target()`, not realizing that it also appends the canvas to the `root_canvas` - but... um... why exactly?